### PR TITLE
Remove unnecessary updateTxFee calls in SendSheet

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-####

## What changed (plus any additional context for devs)

Removing the `gasFeeParamsBySpeed` dep in useEffect which was triggering the gas limit estimation on new params, it was already there before EIP1559 https://github.com/rainbow-me/rainbow/blob/0f845d0bfffb6e9087289c26c101b2cd7bc154ec/src/screens/SendSheet.js#L781 but we don't need it anymore since we're checking for empty params inside the redux action https://github.com/rainbow-me/rainbow/blob/develop/src/redux/gas.ts#L585

## PoW (screenshots / screen recordings)

now the gas limit estimation doesn't depend on new gas price data https://recordit.co/yFFfMC5kap

## Dev checklist for QA: what to test

- gas prices still updating

## Final checklist
[x] Assigned individual reviewers?
[x] Added labels?

### CODE REVIEW

**Comment by estebanmino** [comment]
@brunobar79 pointed out an edge case where the gas limit was calculated but not saved in state because the gas fees weren't ready, this could produce that there were no updates to the fees until the next gasLimit calculation and therefore making the app show values with the default gas limit instead during all the time until the next update. So I added  an action to store the gasLimit for this case.

File: src/redux/gas.ts [hunk 1]
```diff
@@ -584,37 +585,45 @@ export const gasUpdateTxFee = (
...
-    gasFeeParamsBySpeed,
-    _gasLimit,
-    nativeCurrency,
-    _selectedGasFeeOption,
-    txNetwork,
-    l1GasFeeOptimism
-  );
+  ) {
+    // if fee prices not ready, we need to store the gas limit for future calculations
+    // // the rest is as the initial state value
```
Comment (brunobar79) [inline src/redux/gas.ts:590]: extra //

**Review by brunobar79 — APPROVED** [review]
LGTM!

**Review by jinchung — APPROVED** [review]
🌮
